File: app/schemas/student_access_link.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.student import StudentReadDetailed
from app.schemas.user import UserShort


class StudentAccessLinkCreate(BaseModel):
    student_id: int = Field(..., gt=0)
    label: str | None = Field(default=None, max_length=255)
    expires_at: datetime | None = None
    channel: str = Field(default="web", max_length=30)


class StudentAccessLinkRead(BaseModel):
    id: int
    student_id: int
    channel: str
    label: str | None
    issued_by_user_id: int | None
    is_active: bool
    expires_at: datetime | None
    last_used_at: datetime | None
    revoked_at: datetime | None
    usage_count: int
    last_device_id: str | None
    last_device_label: str | None
    created_at: datetime
    student: StudentReadDetailed | None = None
    issued_by_user: UserShort | None = None

    class Config:
        from_attributes = True


class StudentAccessLinkCreateResponse(BaseModel):
    item: StudentAccessLinkRead
    raw_access_token: str

File: frontend/src/types/studentAccessLink.ts
export type StudentAccessLinkItem = {
  id: number
  student_id: number
  channel: string
  label: string | null
  issued_by_user_id: number | null
  is_active: boolean
  expires_at: string | null
  last_used_at: string | null
  revoked_at: string | null
  usage_count: number
  last_device_id: string | null
  last_device_label: string | null
  created_at: string
  student?: {
    id: number
    full_name: string
    group?: { id: number; name: string } | null
  } | null
  issued_by_user?: {
    id: number
    username: string
    full_name: string | null
    role: 'admin' | 'practice_supervisor' | 'viewer'
    is_active: boolean
  } | null
}

export type StudentAccessLinkCreatePayload = {
  student_id: number
  label?: string | null
  expires_at?: string | null
  channel?: string
}

export type StudentAccessLinkCreateResponse = {
  item: StudentAccessLinkItem
  raw_access_token: string
}